Platform/ElkhartlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1884,7 +1884,19 @@ UpdateOsBootMediumInfo (
   OUT  OS_BOOT_OPTION_LIST  *OsBootOptionList
 )
 {
-  FillBootOptionListFromCfgData (OsBootOptionList);
+  OS_BOOT_OPTION             *BootOption;
+  UINT32                      Idx;
+
+  for (Idx = 0; Idx < MAX_BOOT_OPTION_CFGDATA_ENTRY; Idx++) {

```

Comment:
This patch modified CFGDATA directly.  It is not expected.
Also FillBootOptionListFromCfgData() should be outside of the loop.
Please follow Guo's sample.
1. Call  FillBootOptionListFromCfgData() first
2. Do a loop for OsBootOptionList->OsBootOptionCount, and clear BOOT_FLAGS_PREOS if hardware cannot support.

Reply:
Updated with the correct logic, thanks for the review.